subgraphs/stargate/src/sdk/protocols/bridge/index.ts

```diff
@@ -0,0 +1,38 @@
+import { BigInt, ethereum } from "@graphprotocol/graph-ts";
+
+import { AccountManager } from "./account";
+import { Bridge } from "./protocol";
+import { PoolManager } from "./pool";
+import { TokenPricer } from "../config";
+import { BridgeConfigurer } from "./config";
+import { TokenManager, TokenInitializer } from "./tokens";
+
+export class SDK {
+  Protocol: Bridge;
+  Accounts: AccountManager;
+  Pools: PoolManager;
+  Tokens: TokenManager;
+  Pricer: TokenPricer;
+
+  constructor(
+    config: BridgeConfigurer,
+    pricer: TokenPricer,
+    tokenInitializer: TokenInitializer,
+    block: ethereum.Block,
+    transaction: ethereum.Transaction,
+    logIndex: BigInt
```

Comment:
Didn’t it work with `ethereum.Event`? Since block, transaction and logIndex are already included there :P 

I’d like not to change the API of the SDK in a breaking way so that if at any point some functionality is changed or bug fixed we can just replace it everywhere with the new version with no further changes on the mapping.

If necessary, we can change it, but then can you change it in the `sdk` too?

Reply:
In some cases I'm tracking calls, not events. Hence had to change the API of SDK to take block, transaction and logindex separately to work for both calls and events.

If you think this should be part of reference sdk, and not handled on requirement basis, I'll change it there.

Reply:
Oh! that’s a good point, hadn’t thought about that. If you could update it there too it would be great!

I’d suggest creating some custom type that contains all those fields, including an `ethereum.Event | null` (even if redundant). The calls to `onCreatePool` pass an event to the callback, so we might want to keep it.